packages/loopring_v3/contracts/core/impl/LoopringV3.sol

```diff
@@ -159,17 +160,50 @@ contract LoopringV3 is ILoopringV3, ReentrancyGuard
     function updateSettingsInternal(
         address payable  _protocolFeeVault,
         address _blockVerifierAddress,
-        uint    _forcedWithdrawalFee
+        uint    _forcedWithdrawalFee,
+        uint _nextEffectiveTime
         )
         private
     {
         require(address(0) != _protocolFeeVault, "ZERO_ADDRESS");
         require(address(0) != _blockVerifierAddress, "ZERO_ADDRESS");
+        require(
+            _forcedWithdrawalFee <= 0.5 ether,
+            "FORCED_WITHDRAWAL_FEE_TOO_HIGH"
+        );
+
+        if (_nextEffectiveTime == 0) {
+            // effect immediately
+            protocolFeeVault = _protocolFeeVault;
+            blockVerifierAddress = _blockVerifierAddress;
+            forcedWithdrawalFee = _forcedWithdrawalFee;
+        } else {
+            // delayed effect
+            cachedSettings = CachedSettings({
+                protocolFeeVault: _protocolFeeVault,
+                blockVerifierAddress: _blockVerifierAddress,
+                forcedWithdrawalFee: _forcedWithdrawalFee
+            });
+            nextEffectiveTime = _nextEffectiveTime;
+        }
+
+        emit SettingsUpdated(block.timestamp, _nextEffectiveTime);
```

Comment:
should not modify current event, can add another event which only update but not effect